src/type_.rs

```diff
@@ -277,7 +277,10 @@ pub struct Module {
 
 #[derive(Debug, Clone, PartialEq)]
 pub enum PatternConstructor {
-    Record { name: String },
+    Record {
+        name: String,
+        field_map: Option<FieldMap>,
```

Comment:
Uses FieldMap, not &FieldMap to match what is already done in value constructor.